enhancements/monitoring/alerts-ui-management.md

```diff
@@ -0,0 +1,638 @@
++-----------------------+                    +---------------------------+
+                                                       ^
+                                                       |
+                                                       v
+                                            +---------------------------+
+                                            | Console backend / gateway |
+                                            +---------------------------+
+                                                       ^               ^
+                                                       |                \
+                                                       v                 \
+                                            +---------------------------+ \
+                                            |      Alerting API         |  \
+                                            |   /api/v1/alerting        |   \
+                                            +---------------------------+    \
+                                (merges AlertRelabelConfig with Alerts/Rules) \
+                                               ^           ^                   \
+                                              /            |                    \
+                                             v             v                     v
+                  +-------------------------+  +-----------------------+  +----------------------+
+                  | Kubernetes API Server   |  | Thanos/Prometheus API |  | Alertmanager         |
+                  +-------------------------+  +-----------------------+  +----------------------+
+                        ^          ^                       ^                          ^
+                   CRUD |     CRU  |                       |                          |
+                        v          v                       v                          v
+               PrometheusRule   AlertRelabelConfig   Alerts/Rules            Silences/Notifications
+
+
+   Note: Alerts delivery: Prometheus (post‑relabel via AlertRelabelConfig) ---> Alertmanager
+```
+
+Key flows:
+- Web clients authenticate and access the OpenShift Console UI.
+- Console calls the Alerting API to list/manage alerts and rules.
+- Alerting API aggregates rule definitions and alert instances with `AlertRelabelConfig` to present a post‑relabel view (labels, disabled status, effective severity) in list/detail endpoints.
+- Alerting API reads/writes `PrometheusRule` and `AlertRelabelConfig` via the Kubernetes API.
+- Alerting API queries alerts/rules via Thanos/Prometheus.
+
+Rationale for server‑side aggregation:
+- Centralizes a canonical, post‑relabel “effective view” and transactional write path: computes the merged state once for all clients and mediates validated, conflict‑aware updates to `PrometheusRule` and `AlertRelabelConfig`, preventing drift and duplicated logic across UIs.
```

Comment:
> computes the merged state once for all clients

I don't see how it could work since the result depends on the RBAC permissions of individual clients.